### PR TITLE
Better file naming

### PR DESCRIPTION
**Pull request by jakearchibald**
This means any file like `index.js` and `styles.css` will take their final 'name' from the parent directory name.

### CODE REVIEW

**Comment by jakearchibald** [comment]
Before:

```
about/
apple-touch-icon-6d4d3a0c.png
browserify-1c6d72ba.svg
code-splitting/
faqs/
favicon-16x16-a8936e58.png
favicon-32x32-37b750c5.png
favicon-806b6f16.ico
hashing/
importing-modules/
index-003fcd56.css
index-0078a58e.js
index-7032f2c4.js
index-d405582d.js
index.html
labcoat-3c247464.svg
lamp-9f1871d6.svg
non-js-resources/
output-module-formats/
parcel-6852ea4a.svg
plant-27f9dae5.svg
rollup-b67304fe.svg
safari-pinned-tab-4ab9407e.svg
styles-8b5d12f0.css
styles-9d40c531.css
styles-a9532d55.css
styles-e1007afb.css
tooling-report-social-7db67c7c.png
transformations/
walker-7aff6d99.svg
webdev-f3b32dcd.svg
webpack-0395d5ed.svg
```

After:

```
about/
about-17b61a74.css
analytics-ae5574a2.js
apple-touch-icon-6d4d3a0c.png
browserify-1c6d72ba.svg
code-splitting/
faqs/
faqs-416f742a.css
favicon-16x16-a8936e58.png
favicon-32x32-37b750c5.png
favicon-806b6f16.ico
hashing/
importing-modules/
index-003fcd56.css
index-726c3719.css
index-d405582d.js
index.html
labcoat-3c247464.svg
lamp-9f1871d6.svg
non-js-resources/
output-module-formats/
parcel-6852ea4a.svg
plant-27f9dae5.svg
rollup-b67304fe.svg
safari-pinned-tab-4ab9407e.svg
test-951444bb.js
test-b2b0d621.css
tooling-report-social-7db67c7c.png
transformations/
walker-7aff6d99.svg
webdev-f3b32dcd.svg
webpack-0395d5ed.svg
```

File: rollup.config.js [hunk 1]
```diff
@@ -32,6 +33,18 @@ function resolveFileUrl({ fileName }) {
...
 
+const dir = '.tmp/build';
+const staticPath = 'static/[name]-[hash][extname]';
+const jsPath = staticPath.replace('[extname]', '.js');
+
+function jsFileName(chunkInfo) {
+  const parsedPath = parsePath(chunkInfo.facadeModuleId);
+  if (parsedPath.name !== 'index') return jsPath;
+  // Come up with a better name than 'index'
+  const name = parsedPath.dir.split('/').slice(-1);
```
Comment (developit) [inline rollup.config.js:44]: Need to protect against this being the cwd:

```suggestion
  const name = parsedPath.dir.split('/').slice(-1) || 'index';
```

Comment (jakearchibald) [reply rollup.config.js:44]: I'm not sure this is needed. The path is the full path on disk, so it shouldn't be falsey. For scripts that are in one of the roots, the name will end up as "client" or "shared", which seems ok.

Comment (developit) [reply rollup.config.js:44]: ah good point, as long as its not the parent directory name then it's at least deterministic.

**Review by developit — APPROVED** [review]

**Pull request merged**